libgnucash/engine/gnc-commodity.cpp

```diff
@@ -2184,8 +2158,9 @@ gnc_commodity_table_delete_namespace(gnc_commodity_table * table,
     g_list_free(ns->cm_list);
     ns->cm_list = nullptr;
 
-    g_hash_table_foreach_remove(ns->cm_table, ns_helper, nullptr);
-    g_hash_table_destroy(ns->cm_table);
+    std::for_each (ns->cm_table.begin(), ns->cm_table.end(),
+                   [](auto it){ gnc_commodity_destroy (it.second); });
+    StrCommodityMap().swap(ns->cm_table);
```

Comment:
Sigh. Not your fault but this function should call only 
```
{
    gnc_commodity_namespace *ns = gnc_commodity_table_find_namespace(table, name_space);
    g_object_unref(ns);
}
```
The cleanup stuff--removing ns from `table->ns_table` and `table->ns_list` and clearing the StrCommodityMap should be in `gnc_commodity_namespace_dispose_real()`. Call `ns->cm_table.~StrCommodityMap()` in `gnc_commodity_namespace_finalize_real()`.  If you use placement new for constructing the StrCommodityMap then you can call `delete ns->cm_table` instead, which is all-around cleaner. 

There's no point to having GObject memory management if we don't bleeping use it.

Better still of course would be to rewrite `gnc_commodity_namespace` as a proper `GncCommodityNamespace` C++ class, but that's probably more work than you want to do at this point.